api/types/src/address.rs

```diff
@@ -28,7 +28,11 @@ impl FromStr for Address {
     type Err = anyhow::Error;
 
     fn from_str(s: &str) -> anyhow::Result<Self, anyhow::Error> {
-        Ok(Self(AccountAddress::from_hex_literal(s).map_err(|_| {
+        let mut ret = AccountAddress::from_hex_literal(s);
+        if ret.is_err() {
+            ret = AccountAddress::from_hex(s)
+        }
+        Ok(Self(ret.map_err(|_| {
```

Comment:
Are you intending to support both hex literals and non-literals? (leading 0x)

Reply:
Yes.
https://en.wikipedia.org/wiki/Robustness_principle.
I think we should accept all valid account address hex, but only output 0x compact hex-encoded account address.